### PR TITLE
Remove 3 active issues

### LINKED ISSUE #24295: Unix file enumerator isn't thread safe
New test in dotnet/corefx#25426 uncovered this.

```
Unhandled Exception of Type System.NullReferenceException
Message :
System.NullReferenceException : Object reference not set to an instance of an object.
Stack Trace :
   at Interop.Sys.ReadDir(SafeDirectoryHandle dir, DirectoryEntry& outputEntry)
   at System.IO.UnixFileSystem.FileSystemEnumerable`1.<Enumerate>d__11.MoveNext()
   at System.IO.Tests.Regressions.ThreadSafeRepro.Race(IEnumerator`1 s) in /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.IO.FileSystem/tests/Directory/Regressions.cs:line 30
   at System.IO.Tests.Regressions.ThreadSafeRepro.Execute(String directory) in /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.IO.FileSystem/tests/Directory/Regressions.cs:line 53
   at System.IO.Tests.Regressions.FileEnumeratorIsThreadSafe() in /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.IO.FileSystem/tests/Directory/Regressions.cs:line 21
```

### LINKED ISSUE #30108: System.IO.Tests.*.DoesntExistThenCreate_ReturnsDefaultValues fails on CI
- DoesntExistThenCreate_ReturnsDefaultValues

https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38676~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190701.40/workItem/System.IO.FileSystem.Tests

### LINKED ISSUE #43166: System.IO.Tests.File_GetSetTimes.SetDateTimeMax fails on Ubuntu 18.04 in the CI
In two of my unrelated CIs, the System.IO.Tests.File_GetSetTimes.SetDateTimeMax test keeps failing on Ubuntu 18.04 with the following error:
```
Assert.Equal() Failure\nExpected: 9999-04-11T23:47:17.0210000Z\nActual:   2446-05-10T22:38:55.0000000Z
```
There was a notice from the engineering team that they have updated Ubuntu 18.04 image to the latest version, so I wonder if it is related. Setting the file time calls `utimes` or `utimensat` glibc functions, so their implementation w.r.t. clipping the max time could have possibly changed.
<!-- runfo report start -->
Runfo Tracking Issue: [System.IO.Tests.File_GetSetTimes.SetDateTimeMax fails on Ubuntu 18.04 in the CI](https://runfo.azurewebsites.net/tracking/issue/56)
|Build|Definition|Kind|Run Name|
|---|---|---|---|



Build Result Summary
|Day Hit Count|Week Hit Count|Month Hit Count|
|---|---|---|
|0|0|0|

<!-- runfo report end -->

### PR DESCRIPTION
**Pull request by danmoseley**
Fix #30108. remove the test: I cannot see how this test could pass. when you create a directory/file, it does not have default values for its various time attributes.
Fix #24295. We do not plan to make the enumerator thread safe. It happens to be thread safe on Windows, so leaving the test for Windows so that if we do change this for some reason, we can document it in case anyone depends on it.
Fix #43166. Per discussion, this cannot be made to work on modern Linux distros.

### CODE REVIEW

**Comment by danmoseley** [comment]
I was looking through the ActiveIssues on System.IO.FileSystem to get an idea of how much cruft is there. Of 59 ActiveIssue attributes, there are only 12 distinct issues, of which 6 are device/blazor specific and thus relatively new, and of the remaining 6 I looked at 4 and clearly at least these 3 should not exist. They have all been present for years -- one of these issues is over 4 years old. This suggests there's a lot of cruft.

cc @stephentoub @eerhardt as both have pointed out we are over due for cleanup.

**Comment by danmoseley** [comment]
Here's a snippet of one of the issues -- we should Be Bold in such cases:
image

File: src/libraries/System.IO.FileSystem/tests/Directory/EnumerableTests.cs [hunk 1]
```diff
@@ -11,7 +11,10 @@ namespace System.IO.Tests
     public class EnumerableTests : FileSystemTest
     {
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/24295", TestPlatforms.AnyUnix)]
+        // we don't guarantee thread safety of enumerators in general, but on Windows we
+        // currently are thread safe, and this test will help ensure that if we change that
+        // it's a conscious decision. Discussed in #24295.
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/Directory/EnumerableTests.cs:16]:
```suggestion
        // it's a conscious decision. Discussed in https://github.com/dotnet/runtime/issues/24295.
```

File: src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs [hunk 2]
```diff
@@ -148,7 +148,8 @@ public void SetUptoNanoseconds()
             Assert.Equal(ticks, dateTime.Ticks);
         }
 
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/43166", TestPlatforms.Linux)]
+        // Linux kernels no longer have long max date time support. Discussed in #43166.
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs:151]:
```suggestion
        // Linux kernels no longer have long max date time support. Discussed in https://github.com/dotnet/runtime/issues/43166.
```

**Comment by danmoseley** [comment]
@stephentoub

**Review by eerhardt — APPROVED** [review]

**Pull request merged**